src/fromager/resolver.py

```diff
@@ -317,7 +317,7 @@ def get_from_cache(
         cache = self.get_cache()
         # we only want caching for build reqs because for install time reqs we always want to get the latest version
         # we can't guarantee that the latest version is available in the cache so install time reqs cannot use the cache
-        if self.req_type != RequirementType.BUILD:
+        if self.req_type is None or not self.req_type.is_build_requirement:
```

Comment:
I think the comment above this line means that the change to treat installation dependencies of build dependencies as build dependencies may give us different resolutions in builds we're already doing downstream. It's not wrong, but it could be different, so we will want to keep an eye on that.